boa/contracts/vyper/ir_executor.py

```diff
@@ -839,10 +842,11 @@ class Assert(IRExecutor):
 
     def _compile(self, test):
         _ = VMRevert  # make flake8 happy
+        source = self.ir_node.ast_source
         self.builder.extend(
             f"""
         if not bool({test}):
-            VM.vyper_source_pos = {repr(self.ir_node.source_pos)}
+            VM.vyper_source_pos = {repr(source and getpos(source))}
```

Comment:
can we create a small utility which wraps `getpos()` to do this? i'd also like to see a proper branch
```
if source is None:
    return None
return getpos(source)
```